src/ugeneui/src/project_support/ProjectImpl.cpp

```diff
@@ -229,6 +230,21 @@ void ProjectImpl::sl_onObjectRelationChanged(const QList<GObjectRelation> &previ
     }
 }
 
+void ProjectImpl::sl_objectLoadedStateChanged() {
+    Document *doc = qobject_cast<Document *>(sender());
+    CHECK(doc != nullptr, )
+    if (doc->isLoaded()) {
+        return;
+    }
+    foreach (GObject *obj, doc->getObjects()) {
+        UnloadedObject *uo = qobject_cast<UnloadedObject *>(obj);
+        CHECK(obj != nullptr, )
+        GObjectReference from(uo->getDocument()->getURLString(), uo->getGObjectName(), uo->getLoadedObjectType());
+        //invalidate entity ref
+        updateObjectRelations(from, from);
```

Comment:
the call
updateObjectRelations(const GObjectReference &oldRef, const GObjectReference &newRef) 
where both old and new refs are the same looks meaningless.

If it has some 'hidden' behavior could you please be more explicit here? For example create a dedicated method in ProjectImpl to handle this behavior?

Reply:
Next question: if you remove relation on unload, why don't you restore them back on load?